yzerr0/members tab

yzerr0 opened the pull request.

yzerr0 committed:
feat(members_tab): Guild Members Tab Card creation
- created a card component for each group member in members tab
- contains a profile picture and an additional features button

yzerr0 committed:
Merge branch 'main' of https://github.com/cppsea/icebreak into yzerr0/members_tab

nappalion commented on packages/app/components/MemberCard/MemberCard.js:
```diff
@@ -0,0 +1,60 @@
+    },
+});
+
+const MemberCard = ({ name }) => {
```
I like to use the props parameter like in the link below:
https://reactnative.dev/docs/props

nappalion commented on packages/app/components/MemberCard/RoundedIcon.js:
Not sure if we should move this into the general components folder. Make your best judgement (it's fine if we leave it this way).

nappalion commented on packages/app/components/MemberCard/RoundedIcon.js:
```diff
@@ -0,0 +1,35 @@
+import PropTypes from "prop-types";
+import { Image, View } from "react-native";
+
+const RoundedIcon = ({ image }) => {
```
I like to use the props parameter like in the link below:
https://reactnative.dev/docs/props

nappalion commented on packages/app/screens/group/tabs/MembersScreen.js:
```diff
@@ -1,8 +1,51 @@
+    <View>
+      {mockData.map((section) => (
+        <View key={section.key}>
+          <MemberCard name={section.name.length <= 12 ? section.name : section.name.substring(0, 12) + "..."} />
```
Really great solution for now! Just a comment, but we'll have to figure out another way to display user names in the future. Our UI/UX members may be able to come up with a better solution or we may need to limit username length.
https://ux.stackexchange.com/questions/37616/should-a-username-be-truncated/

nappalion commented on packages/app/components/MemberCard/RoundedIcon.js:
```diff
@@ -0,0 +1,35 @@
+};
+
+return (
+    <View style={roundedIconStyle}>
+        <Image
+        style={imageStyle}
+        source={image ? { uri: image } : require("@app/assets/no-pfp-icon.png")}
+        />
+    </View>
```
Could you add click functionality to this icon? It should be a prop, so we can change the functionality from anywhere.

nappalion commented on packages/app/components/MemberCard/MemberFunctions.js:
Like your RFC has, we should add test buttons for now (like block and report) in this "Alert" menu to see how they look (these buttons can then just have (console.log("blocked") or console.log("reported")

nappalion commented on packages/app/components/MemberCard/MemberCard.js:
In your RFC you mentioned click functionality. Could you make it so that clicking a member's card navigates the user to a Dummy Profile tab (research on React Navigation)?

Also you can try using the useNavigation hook. It let's you use navigation from anywhere inside the NavigationStack you're in (lemme know if you have any questions):
https://reactnavigation.org/docs/use-navigation/

nappalion commented on packages/app/components/MemberCard/MemberFunctions.js:
```diff
@@ -0,0 +1,31 @@
+            style = {styles.button}
+            onPress={() => Alert.alert("Three dots button pressed")}
+        >
+            <Text style={styles.buttonText}>...</Text>
```
Could you align this vertically?

nappalion commented on packages/app/components/MemberCard/RoundedIcon.js:
```diff
@@ -0,0 +1,35 @@
+import { Image, View } from "react-native";
+
+const RoundedIcon = ({ image }) => {
+    const roundedIconStyle = {
+        width: 60,
+        height: 60,
+        borderRadius: 30,
+        justifyContent: "center",
+        alignItems: "center",
+        overflow: "hidden",
+    };
+
+const imageStyle = {
+    aspectRatio: 1,
+    width: "100%",
+    height: "100%",
+    borderRadius: 15,
+};
```
This should be in a styles variable (StyleSheet).

nappalion reviewed (commented):
Really great work! Your code (and the UI) is very clean. I've left some suggestions regarding changes mentioned in your RFC (and also some minor ones). Let me know if you have any questions.

yzerr0 committed:
Merge branch 'main' of https://github.com/cppsea/icebreak into yzerr0/members_tab

MichaelWuhu committed:
Merge branch 'main' of https://github.com/cppsea/icebreak into yzerr0/members_tab

yzerr0 committed:
Merge branch 'main' of https://github.com/cppsea/icebreak into yzerr0/members_tab

yzerr0 committed:
feat: clickable icons and made three dots vertical

MichaelWuhu committed:
chore: updated according to PR comments
-added prop validation to components
-added MemberFunctions report and block to print to console

yzerr0 committed:
chore: merge=

yzerr0 committed:
Merge branch 'main' of https://github.com/cppsea/icebreak into yzerr0/members_tab

MichaelWuhu committed:
Merge remote-tracking branch 'origin/main' into yzerr0/members_tab

MichaelWuhu committed:
fix: react-native-option-menu not installed with yarn
- ran: yarn add react-native-option-menu and these were the changes

lxkedinh committed:
feat: created guild context

MichaelWuhu committed:
Merge branch 'yzerr0/members_tab' of https://github.com/cppsea/icebreak into yzerr0/members_tab

MichaelWuhu committed:
feat: used GuildContext to display member names
- also added "Cancel" option to member functions

yzerr0 committed:
chore: merge branch on repo

yzerr0 committed:
Merge branch 'main' of https://github.com/cppsea/icebreak into yzerr0/members_tab

yzerr0 committed:
feat: added dummy profile popup on icon click

MinT-Napkin commented on packages/app/screens/group/tabs/MembersScreen.js:
```diff
@@ -1,8 +1,27 @@
+                : (member.firstName + " " + member.lastName).substring(0, 12) +
+                  "..."
+            }
+          />
```
Just a minor thing to fix/add in. Icons don't appear for some of the members that have avatars in the database and just has the no-pfp-icon by default, just need to add in the avatar image from the guild context into the MemberCard and you're good to go!

MinT-Napkin reviewed (approved):
Lgtm! 

Just a minor change needed in one of the files if you can.

MichaelWuhu committed:
fix: made profile picture display on member card

MichaelWuhu committed:
Merge branch 'yzerr0/members_tab' of https://github.com/cppsea/icebreak into yzerr0/members_tab
